### PR TITLE
image component height fix

### PR DESCRIPTION
**Pull request by joshfarrant**
## Summary

Fixes incorrect `height` of the `Image` component when  `as="picture"`.

## What should reviewers focus on?

- Ensure there are no visual regressions in the `Image` component when used with the `as="picture"` prop. From what I can tell we never actually use that prop in the codebase, but I'd appreciate you verifying this please.

## Supporting resources (related issues, external links, etc):

- https://github.com/primer/brand/issues/604

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

These examples are rendered with the following code

```tsx
<Stack direction="horizontal">
  <Card href="https://github.com" fullWidth hasBorder>
    <Card.Image
      aspectRatio="16:10"
      as="picture"
      height={100}
      src={placeholderImage}
      alt="placeholder, blank area with an gray background color"
    />
    <Card.Heading>GitHub Actions cheat sheet</Card.Heading>
    <Card.Description>Integer pellentesque...</Card.Description>
  </Card>
  <Card href="https://github.com" fullWidth hasBorder>
    <Card.Image
      aspectRatio="16:10"
      as="picture"
      height={100}
      src={placeholderImage}
      alt="placeholder, blank area with an gray background color"
    />
    <Card.Heading>GitHub Actions cheat sheet</Card.Heading>
    <Card.Description>Integer pellentesque...</Card.Description>
  </Card>
</Stack>
```

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

image

 </td>
<td valign="top">

image

</td>
</tr>

</table>

### CODE REVIEW

**Review by rezrah — COMMENTED** [review]
Thanks for the fix @joshfarrant 🙏 

- Could you add a changeset please? Just needs to indicate the `height` prop is now forwarded correctly to underlying elements.
- can we add a test for this please?
- (nice to have) Could we update this example to add an `as="picture` variant alongside the default? or add another story on its own just for picture element.

**Comment by joshfarrant** [comment]
@rezrah One day I'll remember the changeset 🤦‍♂️

Test and changeset added, example to follow shortly

**Review by rezrah — APPROVED** [review]

File: .changeset/three-singers-float.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': patch
+---
+
+In `Image` component, the `height` prop is now correctly forwarded to the underlying `img` element
```
[the remark this answers is not in the document]
Comment (joshfarrant) [reply .changeset/three-singers-float.md:5]: It is technically passed to the `img` element, but if you think that's a clearer changelog it works for me

https://github.com/primer/brand/pull/630/files#diff-328f63a328a810b443cedb144398ffdb929a5bec828d38cebc7dce781c275086R84

[the remark this answers is not in the document]
Comment (rezrah) [reply .changeset/three-singers-float.md]: Yeah, just appended the `when using `as="picture" customization.` bit as that's the important part of the change.

**Pull request merged**